design/baremetal-operator/raid-disk-controller-naming.md

```diff
@@ -0,0 +1,213 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# raid-extension-for-physical-disks-and-controller
+
+## Status
+
+provisional
+
+## Summary
+
+This document provides the naming convention adopted to express
+``physical_disks`` and ``controller`` sections of a RAID configuration of
+a baremetal-host. The aim is to provide a uniform way of representation for
+hardware from various vendors, in a way that enhances operator experience.
+It also proposes an implementation for the same.
+
+## Motivation
+
+Vendors have different ways of defining the names for their physical disks
+and RAID controllers within Ironic. This leads to a complication for the
+operator where user experience is affected. There needs to be a uniform
+naming convention that abstracts this and increases convenience.
+
+### Goals
+
+The primary goal is to reach consensus on one naming convention to depict
+disk and RAID controller names in the baremetal-host ``hardware-raid``
+section.
+
+Afterwards, implementation of the same will be done within the
+``baremetal-operator`` by extending the ``baremetal-host`` specification.
+
+### Non-Goals
+
+This specification does not deal with ``software-raid`` and extension for it.
+
+## Proposal
+
+The crux of the proposal is the naming convention that is to be adopted
+for interfacing with the user. Currently, we have the following options
+
+### Controller Naming
+
+For controllers that are installed specifically for the operating system,
+for example the [BOSS Card](1) by DellEMC, they are almost always used with
+two disks/drives in ``RAID-1``, and can be refered to as
+``primary-controller``. Similar method can be adopted to address such
+controllers by other vendors.
+
+Other controllers that are installed in the server can be used in any order,
+and thus can be named ``secondary-controller-x`` where x can be an integer
+greater than or equal to 1, e-g ``secondary-controller-1``,
+``secondary-controller-2`` and so on.
+
+### Physical Disk/Drive Naming
+
+With physical disks, the primary issue is disk enumeration. We can address
+it this way:
+
+- Start from the front bay, and count drives in the order that they are
+  listed by vendors. For instance, vendors would list them as XXXXX-1-XXX,
+  XXXXX-2-XXX and so on. We can use these 1 and 2 to specify ordering and
+  provide a simpler naming convention as ``Disk-1``, ``Disk-2`` and so on.
```

Comment:
Why not just use the number?

Reply:
I'm fine with using just the number. Let's see if anybody else has any comments.

Reply:
Just use the number unless there is some objection.

Reply:
Do we have any ordering guarantees across disks at least within a controller?

Reply:
Correct me if I'm wrong @rpioso but the ordering of disks _is_ deterministic for iDRAC BMC type. I can't be sure about other vendors since I don't have availability of that kind of hardware. 
Perhaps @Hellcatlk can tell us about RMC type hardware? 

Regardless, I'm of the opinion that the numbering _should_ be deterministic and vendors should bear the burden of this. We can proceed with development and make this determinism a necessary condition.

Reply:
For irmc, the ordering of disks  is also fixed.